crates/rpc/rpc-convert/src/transaction.rs

```diff
@@ -826,7 +847,7 @@ where
         Ok(self
             .sim_tx_converter
             .convert_sim_tx(request)
-            .map_err(|e| TransactionConversionError(e.to_string()))?)
+            .map_err(|_| TransactionConversionError::MissingRequiredFields)?)
```

Comment:
we don't want to loose the error returned by `convert_sim_tx` as it may have valuable info, what error type is that? we should have the `MissingRequiredFields` variant wrap that error type i.e. `MissingRequiredFields(TheErrorType)`